benchadapt/python/benchadapt/result.py

```diff
@@ -59,15 +59,23 @@ class BenchmarkResult:
     stats: Dict[str, Any] = None
     tags: Dict[str, Any] = field(default_factory=dict)
     info: Dict[str, Any] = None
-    machine_info: Dict[str, Any] = None
+    machine_info: Dict[str, Any] = field(
+        default_factory=lambda: machine_info(host_name=None)
+    )
     cluster_info: Dict[str, Any] = None
     context: Dict[str, Any] = None
     github: Dict[str, Any] = field(default_factory=github_info)
     error: str = None
 
-    def __post_init__(self):
-        if not self.machine_info and not self.cluster_info:
-            self.machine_info = machine_info(host_name=None)
+    @property
+    def _cluster_info(self) -> Dict[str, Any]:
+        return self.__cluster_info
```

Comment:
wow yeah this is ugly haha! But I can't think of a better way to avoid overriding the dataclasses magic.

Maybe it would be a little easier to understand if you renamed `_cluster_info` -> `_temp_cluster_property` and `__cluster_info` -> `_cluster_info_cache`? (unless you somehow need the [name mangling](https://docs.python.org/3/tutorial/classes.html#private-variables) of the double underscore)